tests/unit/container/test_with_parents.py

```diff
@@ -0,0 +1,141 @@
+from abc import ABC
+from typing import Generic, NamedTuple, Protocol, TypeVar, TypeVarTuple
+
+from dishka import make_container, Provider, Scope
+from dishka.entities.with_parents import WithParents
+from dishka.exceptions import NoFactoryError
+
+
+def test_simple_inheritance() -> None:
+    class A1: ...
+    class A2(A1): ...
+    class A3(A2): ...
+    
+    provider = Provider(scope=Scope.APP)
+    provider.provide(lambda: 1, provides=WithParents[A3])
```

Comment:
let's check that value generates only once when accessing by multiple classes, so as there are aliases

Reply:
done